lib/src/command/add.dart

```diff
@@ -75,63 +84,121 @@ class AddCommand extends PubCommand {
+      /// conflicts in other parts of [pub] if the pubspec is newer than
+      /// `pubspec.lock`.
+      _updatePubspec(resultPackage, packageInformation, isDev);
+
+      /// Create a new [Entrypoint] since we have to reprocess the updated
+      /// pubspec file.
+      await Entrypoint.current(cache).acquireDependencies(SolveType.GET,
+          precompile: argResults['precompile']);
+    }
 
     if (isOffline) {
       log.warning('Warning: Packages added when offline may not resolve to '
-          'the latest versions of your dependencies.');
+          'the latest compatible version available.');
     }
   }
 
-  /// Creates a new in-memory [Pubspec] by adding [package] to [original].
+  /// Creates a new in-memory [Pubspec] by adding [package] to the
+  /// dependencies of [original].
   Future<Pubspec> _addPackageToPubspec(
       Pubspec original, PackageRange package) async {
     ArgumentError.checkNotNull(original, 'original');
     ArgumentError.checkNotNull(package, 'package');
 
     final dependencies = [...original.dependencies.values];
     var devDependencies = [...original.devDependencies.values];
+    final dependencyNames = dependencies.map((dependency) => dependency.name);
+    final devDependencyNames =
+        devDependencies.map((devDependency) => devDependency.name);
 
-    if (isDevelopment) {
-      final dependencyNames = dependencies.map((dependency) => dependency.name);
+    if (isDev) {
+      /// TODO(walnut): Change the error message once pub upgrade --bump is
+      /// released
+      if (devDependencyNames.contains(package.name)) {
+        usageException('"${package.name}" is already in "dev_dependencies". '
+            'Please use "pub upgrade ${package.name}" if you wish to upgrade '
+            'to a later version!');
+      }
 
       /// If package is originally in dependencies and we wish to add it to
       /// dev_dependencies, this is a redundant change, and we should not
       /// remove the package from dependencies, since it might cause the user's
       /// code to break.
       if (dependencyNames.contains(package.name)) {
-        usageException('${package.name} is already in dependencies. '
-            'Please remove existing entry before adding it to dev_dependencies');
+        usageException('"${package.name}" is already in "dependencies". '
+            'Please use "pub remove ${package.name}" to remove it before adding '
+            'it to "dev_dependencies"');
       }
 
       devDependencies.add(package);
     } else {
-      final devDependencyNames =
-          devDependencies.map((devDependency) => devDependency.name);
+      /// TODO(walnut): Change the error message once pub upgrade --bump is
+      /// released
+      if (dependencyNames.contains(package.name)) {
+        usageException('"${package.name}" is already in "dependencies". '
+            'Please use "pub upgrade ${package.name}" if you wish to upgrade '
```

Comment:
```suggestion
            'Use "pub upgrade ${package.name}" to upgrade '
```

I'm not a messaging expert, but short and concise is usually better.

Similar in other places.. If we can skip `Please` and `if you wish` that might be good.
We can say `Use ... to ...`

Reply:
Removed the `Please` and `if you wish` from everywhere I could find!